integrationtests/subscribe_update_test.go

```diff
@@ -0,0 +1,489 @@
+package integrationtests
+
+import (
+	"context"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/mengelbart/moqtransport"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// TestSubscribeUpdateHandler verifies that SUBSCRIBE_UPDATE messages are properly 
+// passed to the application handler
+func TestSubscribeUpdateHandler(t *testing.T) {
+	// Track subscription updates received by the server
+	type subscriptionUpdate struct {
+		requestID          uint64
+		priority           uint8
+		forward            uint8
+		startGroup         uint64
+		startObject        uint64
+		endGroup           uint64
+	}
+	
+	var updateMu sync.Mutex
+	var receivedUpdates []subscriptionUpdate
+
+	// Create server that handles subscriptions and tracks updates
+	serverHandler := moqtransport.HandlerFunc(func(w moqtransport.ResponseWriter, r *moqtransport.Message) {
+		switch r.Method {
+		case moqtransport.MessageAnnounce:
+			t.Logf("Server: Accepting announcement for namespace %v", r.Namespace)
+			err := w.Accept()
+			require.NoError(t, err)
+			
+		case moqtransport.MessageSubscribe:
+			t.Logf("Server: Accepting subscription for track %s", r.Track)
+			err := w.Accept()
+			require.NoError(t, err)
+			
+		case moqtransport.MessageSubscribeUpdate:
+			t.Logf("Server: Received SUBSCRIBE_UPDATE requestID=%d, priority=%d, forward=%d",
+				r.RequestID, r.SubscriberPriority, r.Forward)
+			
+			updateMu.Lock()
+			receivedUpdates = append(receivedUpdates, subscriptionUpdate{
+				requestID:   r.RequestID,
+				priority:    r.SubscriberPriority,
+				forward:     r.Forward,
+				startGroup:  r.StartGroup,
+				startObject: r.StartObject,
+				endGroup:    r.EndGroup,
+			})
+			updateMu.Unlock()
+		}
+	})
+
+	// Setup client and server connections
+	serverConn, clientConn, done := connect(t)
+	defer done()
+	
+	_, _, _, clientSession, cleanup := setup(t, serverConn, clientConn, serverHandler)
+	defer cleanup()
+	
+	ctx := context.Background()
+	
+	// Announce namespace
+	err := clientSession.Announce(ctx, []string{"test", "namespace"})
+	require.NoError(t, err)
+	
+	// Subscribe to a track
+	track, err := clientSession.Subscribe(ctx, []string{"test", "namespace"}, "video", "")
+	require.NoError(t, err)
+	require.NotNil(t, track)
+	
+	// Note: This test verifies that SUBSCRIBE_UPDATE messages are properly
+	// passed to the application handler when received by the server.
+	// The client can send SUBSCRIBE_UPDATE messages using the Session.SubscribeUpdate
+	// method (see TestClientSubscribeUpdate for an example).
```

Comment:
I think something is missing here. We never use trigger the subscribe update message here and also don't verify the result?